### PR TITLE
HCP Link Status Parsing and Modal Update 

### PR DESCRIPTION
**Pull request by zofskeez**
The return value for `hcp_link_status` changed slightly with an updated timestamp format and formatted error messages. This prompted a design update to display a common message in the banner for disconnected/connecting errors and use of a modal for additional information. The connected state remains unchanged.

image
image
image

### CODE REVIEW

File: ui/app/styles/core/buttons.scss [hunk 1]
```diff
@@ -280,5 +280,4 @@ a.button.disabled {
   font-size: inherit;
   font-weight: inherit;
   cursor: pointer;
-  color: $link;
```
Comment (zofskeez) [inline ui/app/styles/core/buttons.scss:283]: I think this was originally intended to be a sort of reset class to make a button look like the surrounding text and then to use additional classes to change the color or underline for specific scenarios. `color: inherit` was already defined on line 279.

File: ui/app/templates/components/oidc/scope-form.hbs [hunk 2]
```diff
@@ -39,7 +39,7 @@
       You can use Alt+Tab (Option+Tab on MacOS) in the code editor to skip to the next field. See
       <button
         type="button"
-        class="text-button"
+        class="text-button has-text-info"
```
Comment (zofskeez) [inline ui/app/templates/components/oidc/scope-form.hbs:42]: This preserves the styling after removing `color: $link` from the `text-button` class.

File: ui/app/templates/components/link-status.hbs [hunk 3]
```diff
@@ -8,8 +8,59 @@
...
       {{/if}}
     </p>
   </div>
-{{/if}}
\ No newline at end of file
+{{/if}}
+
+{{#if this.isTesting}}
+  <div id="modal-wormhole"></div>
+{{/if}}
```
Comment (zofskeez) [inline ui/app/templates/components/link-status.hbs:23]: This prevents having to add the wormhole to every component test that uses `nav-header`

File: ui/tests/acceptance/transit-test.js [hunk 4]
```diff
@@ -258,7 +258,7 @@ const testConvergentEncryption = async function (assert, keyName) {
     }
     // store ciphertext for decryption step
     const copiedCiphertext = find('[data-test-encrypted-value="ciphertext"]').innerText;
-    await click('[data-test-modal-background]');
+    await click('.modal.is-active [data-test-modal-background]');
```
Comment (hellobontempo) [inline ui/tests/acceptance/transit-test.js:261]: thank you for adding this! the `is-active` part often gets left out of tests..

File: ui/app/styles/core/buttons.scss (hunk 1 above)
Comment (hellobontempo) [reply ui/app/styles/core/buttons.scss:283]: thanks for cleaning up!

**Review by hellobontempo — APPROVED** [review]
Thanks for handling all this back and forth! Nice work. 🚀

**Pull request merged**